synapse/storage/data_stores/main/events_bg_updates.py

```diff
@@ -503,3 +508,60 @@ def _event_fix_redactions_bytes_txn(txn):
         yield self._end_background_update("event_fix_redactions_bytes")
 
         return 1
+
+    @defer.inlineCallbacks
+    def _event_store_labels(self, progress, batch_size):
+        """Stores labels for events."""
+        last_event_id = progress.get("last_event_id", "")
+
+        def _event_store_labels_txn(txn):
+            txn.execute(
+                """
+                SELECT event_id, json FROM event_json
+                LEFT JOIN event_labels USING (event_id)
+                WHERE event_id > ? AND label IS NULL
+                ORDER BY event_id LIMIT ?
+                """,
+                (last_event_id, batch_size),
+            )
+
+            rows = self.cursor_to_dict(txn)
+            if not rows:
+                return True, 0
+
+            for row in rows:
```

Comment:
I feel like this might be simpler (and require less memory) without `cursor_to_dict`:

```
txn.execute(...)
rowcount=0
for (event_id, event_json) in txn:
    ...
    rowcount += 1

if rowcount > 0:
   # update progress

return rows
```

ymmv though. it doesn't matter too much.

Reply:
Oh, I didn't know it could be used like this! Yeah this would be much simpler.